specifyweb/frontend/js_src/lib/components/TreeView/Row.tsx

```diff
@@ -175,9 +177,11 @@ export function TreeRow({
                 color:
                   typeof row.acceptedId === 'number' ? synonymColor : undefined,
               }}
-              onClick={({ metaKey, shiftKey }): void =>
-                metaKey || shiftKey ? handleFocusNode([]) : handleToggle(false)
-              }
+              onClick={({ metaKey, shiftKey }): void => {
+                metaKey || shiftKey
+                  ? handleFocusNode([])
+                  : (handleToggle(false), setLastFocusedTree());
```

Comment:
this is hacky (avoid comma expressions as they can create bug prone and unreadable code), and buggy - setLastFocusedTree is only called on mouse click, not onKeyDown - why not call `setLastFocusedTree` inside of handleToggle instead?

Reply:
but EVEN BETTER, why not call setLastFocusedTree() when `setFocusPath` is called in `const treeContainer = ...` in `index.tsx`?

that way, this file won't need to be changed at all - to keep code cleaner and decoupled this file should not know anything about the fact that multiple trees can be displayed as that is above it - this file is only concerned with displaying a single tree row, and calling some callbacks on interactions